ortools/linear_solver/xpress_interface.cc

```diff
@@ -1209,16 +1227,162 @@ void XpressInterface::SetLpAlgorithm(int value) {
   }
 }
 
+bool XpressInterface::readParameter(std::string const &name, std::string const &value)
+{
+  if (!value.size()) {
+    LOG(DFATAL) << "Empty value for parameter '" << name << "' in "
+                << SolverVersion();
+    return false;
+  }
+  int id, type;
+  if (XPRSgetcontrolinfo(mLp, name.c_str(), &id, &type) ||
+      type == XPRS_TYPE_NOTDEFINED)
+  {
+    LOG(DFATAL) << "Unknown parameter '" << name << "' in "
+                << SolverVersion();
+    return false;
+  }
+  std::stringstream v(value);
+  switch (type) {
+  case XPRS_TYPE_INT:
+  {
+    int i;
+    v >> i;
+    if (!v.eof()) {
+      LOG(DFATAL) << "Failed to parse value '" << value
+                  << "' for int parameter '" << name << "' in "
+                  << SolverVersion();
+      return false;
+    }
+    if (XPRSsetintcontrol(mLp, id, i)) {
+      LOG(DFATAL) << "Failed to set int parameter '" << name
+                  << "' to " << value << " (" << i << ") in "
+                  << SolverVersion();
+      return false;
+    }
+  }
+  break;
+  case XPRS_TYPE_INT64:
+  {
+    XPRSint64 i;
+    v >> i;
+    if (!v.eof()) {
+      LOG(DFATAL) << "Failed to parse value '" << value
+                  << "' for int64 parameter '" << name << "' in "
+                  << SolverVersion();
+      return false;
+    }
+    if (XPRSsetintcontrol64(mLp, id, i)) {
+      LOG(DFATAL) << "Failed to set int64 parameter '" << name
+                  << "' to " << value << " (" << i << ") in "
+                  << SolverVersion();
+      return false;
+    }
+  }
+  break;
+  case XPRS_TYPE_DOUBLE:
+  {
+    double d;
+    v >> d;
+    if (!v.eof()) {
+      LOG(DFATAL) << "Failed to parse value '" << value
+                  << "' for dbl parameter '" << name << "' in "
+                  << SolverVersion();
+      return false;
+    }
+    if (XPRSsetdblcontrol(mLp, id, d)) {
+      LOG(DFATAL) << "Failed to set double parameter '" << name
+                  << "' to " << value << " (" << d << ") in "
+                  << SolverVersion();
+      return false;
+    }
+  }
+  break;
+  default:
+    // Note that string parameters are not supported at the moment since
+    // we don't want to deal with potential encoding or escaping issues.
+    LOG(DFATAL) << "Unsupported parameter type " << type
+                << " for parameter '" << name << "' in "
+                << SolverVersion();
+    return false;
+  }
+
+  return true;
+}
+
+bool XpressInterface::readParameters(std::istream &is, char sep) {
+  // - parameters must be specified as NAME=VALUE
+  // - settings must be separated by sep
+  // - any whitespace is ignored
+  // - string parameters are not supported
+
+  std::string name(""), value("");
+  bool inValue = false;
```

Comment:
You may use [`absl::StrSplit`](https://abseil.io/docs/cpp/guides/strings#abslstrsplit-for-splitting-strings) here.